### PR TITLE
API Namespace origin

### PR DESCRIPTION
**Pull request by zdne**
Ready to be reviewed & merged. In a short term future we may want to add HOST -specific element or a tooling metadata but lets leave it out for now.

### CODE REVIEW

File: namespaces/api-namespace.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            - scenario - Reserved. Category is set of steps.
+- `content` (array[Element])
+
+### Example
+
+```json
+{
+    "element": "category",
+    "meta": {
+        "class": "api",
```
Comment (smizell) [inline namespaces/api-namespace.md]: I may have missed this on other PRs, but class is defined as an array of strings in the spec.

File: namespaces/api-namespace.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+content.
+
+The Copy element MAY appear as a content of any element defined in the base
+namespaces.
+
+### Properties
+
+- `element`: copy (string, fixed)
+- `attributes` (object)
+    - `contentType` (string) - Optional media type of the content E.g. text/html or text/plain.
```
Comment (smizell) [inline namespaces/api-namespace.md]: Good idea

File: namespaces/api-namespace.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+If a Copy element the parent's element `description` metadata MAY include its
+content.
+
+The Copy element MAY appear as a content of any element defined in the base
+namespaces.
+
+### Properties
+
+- `element`: copy (string, fixed)
```
Comment (smizell) [inline namespaces/api-namespace.md:88]: I'm wondering about this name. It has a very verb-sounding name. I'm not against it, just didn't immediately get what it was.

File: namespaces/api-namespace.md (hunk 1 above)
Comment (zdne) [inline namespaces/api-namespace.md]: Oh that is my oversight here – the element definitions at line 39 is array. Will fix the example. thanks

File: namespaces/api-namespace.md (hunk 3 above)
Comment (zdne) [inline namespaces/api-namespace.md:88]: No, it is a noun. Not verb :)  http://en.wikipedia.org/wiki/Copy_(written)

Comment (zdne) [inline namespaces/api-namespace.md:88]: Remember it may include images – through links (HTML :)

Comment (smizell) [inline namespaces/api-namespace.md:88]: > No, it is a noun. Not verb :)

Right. I was mostly given my first impression. Do with that as you see fit :)

**Comment by smizell** [comment]
@zdne Let me know when fixed. Should be good to go once example is updated.

File: namespaces/api-namespace.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+}
+```
+
+## Copy (Element)
+
+Copy element represents a copy text. A textual information in API description.
+Its content is a string and it MAY include information about the media type
+of the copy's content.
+
+If a Copy element the parent's element `description` metadata MAY include its
```
Comment (smizell) [inline namespaces/api-namespace.md]: Is this sentence correct?

File: namespaces/api-namespace.md (hunk 2 above)
Comment (smizell) [inline namespaces/api-namespace.md]: Any benefit in making this a sample rather than just in the description?

**Comment by smizell** [comment]
@zdne Added a couple of extra comments.

File: namespaces/api-namespace.md (hunk 2 above)
Comment (zdne) [inline namespaces/api-namespace.md]: @smizell Not sure what you mean by sample?

Comment (smizell) [inline namespaces/api-namespace.md]: Something like this:
https://github.com/apiaryio/mson/blob/master/MSON%20Specification.md#44-sample

Comment (zdne) [inline namespaces/api-namespace.md]: Oh got it. Will fix it :)

File: namespaces/api-namespace.md (hunk 4 above)
Comment (zdne) [inline namespaces/api-namespace.md]: It isn't! Thanks

File: namespaces/api-namespace.md [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+`dataStructures` to denote it includes both resource and data structures.
+
+### Properties
+
+- `element`: category (string, fixed)
+- `meta`
+    - `class` (array, fixed)
+        - (enum[string])
+            - api - Category is a API top-level group.
+            - resourceGroup - Category is a set of resource.
```
Comment (pksunkara) [inline namespaces/api-namespace.md:42]: @zdne Didn't we decide to remove the concept of `resource group`?

Comment (zdne) [inline namespaces/api-namespace.md:42]: @pksunkara yes, that why this is a `category` element – a generic grouping element, not a `resource group` element. `class` serves just an additional classification of the element, but the semantically important information here is `category` – any tool that knows how to deal with a category should be able to understand the element, some tools, can use the additional taxonomy of the element to provide some finer features, but in reality it should be really about `category`

Comment (pksunkara) [inline namespaces/api-namespace.md:42]: Okay, thought so, but just wanted to confirm.

**Pull request merged**